### ISSUE TITLE
SubViewport texture on mesh doesn't show on exported build unless render target set to update always

### ISSUE THREAD

**Issue by elvisish** [godotengine/godot#86258]
### Tested versions

4.2.1

### System information

Windows 10, Vulkan, RTX 3060

### Issue description

SubViewport render textures give the error in editor:
```swift
_on_start_button_down(): Viewport Texture must be set to use it.
```
but display correctly in-game when run from the editor. Exported builds, however, totally break and just show black textures unless the render mode is set to Always.

https://github.com/godotengine/godot/issues/55471 addressed it, but was closed. It is still an issue, but hopefully these steps are useful enough to reproduce.

### Steps to reproduce

- Add a viewport material to a mesh and add a child Subviewport node
- set render target to something other than Always
- see the texture works in the game run from the editor
- export and see the texture is black in export build

### Minimal reproduction project (MRP)

N/A as this is a popular enough bug without needing me to make an example.

**Comment by EvilRacer329** [comment]
Have a reliable replication of this issue. The error doesn't stop it running perfectly in debug mode, but exports won't run. ONly workaround is to remove the ViewportTexture from the object.

**Comment by elvisish** [comment]
> Have a reliable replication of this issue. The error doesn't stop it running perfectly in debug mode, but you get the error every time you open the scene. Only workaround is to remove the ViewportTexture from the object.

Your exports don't run? I haven't had that problem, although some folks had issues with my Linux builds and this error appears on every, single, log so maybe it is causing some issues with running builds.

**Comment by EvilRacer329** [comment]
> > Have a reliable replication of this issue. The error doesn't stop it running perfectly in debug mode, but you get the error every time you open the scene. Only workaround is to remove the ViewportTexture from the object.
> 
> Your exports don't run? I haven't had that problem, although some folks had issues with my Linux builds and this error appears on every, single, log so maybe it is causing some issues with running builds.

No, it was my mistake - I'm still getting the hang of version control and I'd made another change that was breaking exports but not debug runs. Unrelated. :)

**Comment by Rindbee** [comment]
The error message is a separate known issue if it appears during scene instantiation.

During the instantiation of a scene, the instantiation of resources occurs before the instantiation of nodes. This may cause problems setting up resources with `resource_local_to_scene` enabled (especially `ViewportTexture`, it needs the target viewport's information).


https://github.com/godotengine/godot/blob/0bcc0e92b3f0ac57d4c4650722f347593a258572/scene/resources/material.cpp#L1936-L1939

Usually occurs when `ViewportTexture` is used as `albedo_texture` of material.

**Comment by jamie-pate** [comment]
I struggled with this for a long time until I realized that `local to scene` was not checked in my mesh and/or material.. if this is not checked the viewport texture will never call `set_local_to_scene` which is where it actually checks the viewport path and hooks up the texture for the scene.
